modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1211,10 +1211,10 @@ private CompletableFuture<?> onTableRename(RenameTableEventParameters parameters
                         return failedFuture(e);
                     }
 
-                    TableImpl table = tables.get(parameters.tableId());
+                    TableViewInternal table = tables.get(parameters.tableId());
 
                     // TODO: revisit this approach, see https://issues.apache.org/jira/browse/IGNITE-21235.
-                    table.name(parameters.newTableName());
+                    ((TableImpl) table).name(parameters.newTableName());
```

Comment:
This looks flaky. Just an observation, not a blocker IMO if the intention is to revisit after IGNITE-21235.